src/ui/Filters/Filters.js

```diff
@@ -0,0 +1,203 @@
+      animation-duration: ${animationsMillisecondsDuration}ms;
+      animation-name: ${animationClosing(filtersAnimationClose)};
+    }
+    &.mobile-closed {
+      display: none;
+      height: 0px;
+      margin-top: -35px;
+      opacity: 0;
+    }
+    &.mobile-opening {
+      animation-duration: ${animationsMillisecondsDuration}ms;
+      animation-name: ${animationOpening(filtersAnimationOpen)};
+    }
+    &.mobile-open {
+      height: 100%;
+      margin-top: 6px;
+      opacity: 1;
+    }
+  }
+  @media ${props => props.theme.queries.desktop} {
+    &.mobile-open,
+    &.mobile-closed {
+      height: 100%;
+      margin-top: 6px;
+      opacity: 1;
+    }
+  }
+  `;
+
+const ButtonContainer = styled.div`
+  width: 70px;
+  align-self: flex-end;
+  margin-left: auto;
+  @media ${props => props.theme.queries.desktop} {
+    width: 200px;
+  }
+`;
+
+const ClearButton = styled(Button)`
+  text-align: right;
+`;
+  
+class Filters extends React.Component {
+  
+  state = {
+    openFilter: null,
+  };
+    
+  static propTypes = {
+    title: PropTypes.string.isRequired,
+    options: PropTypes.arrayOf(PropTypes.shape({}).isRequired).isRequired,
+    activeFilters: PropTypes.shape({}).isRequired,
+    clearFilters: PropTypes.func.isRequired,
+    setFilter: PropTypes.func.isRequired,
+    children: PropTypes.node.isRequired,
+  };
+
+  static defaultProps = {
+    activeFilters: {},
+  };
+
+  _handleFilterTouch = (filter) => {
+    this.setState(({ openFilter }) => ({
+      openFilter: filter === openFilter ? null : filter,
+      forceCollapse: false
+    }));
+  };
+
+  _clearFilters = () => {
+    this.setState({ forceCollapse: true }, () => {
+      this.props.clearFilters();
+    })
+  }
+
+  render() {
+    const {
+      title,
+      options,
+      activeFilters,
+      setFilter,
+      children,
+    } = this.props;
+
+    return (
+      <FiltersContainer>
+        <Header>
+          <Title>
+            {title}
+          </Title>
+          <ButtonContainer>
+            <ClearButton
+              transparent
+              color="terracota"
+              variant="primary"
+              uppercase={false}
+              onClick={this._clearFilters}
+              spacing="0"
+              height="20px"
+              weight="normal"
+            >
+              Clear All
+            </ClearButton>
+          </ButtonContainer>
+        </Header>
+        {React.Children.map(children, (child) => {
+          const { accessor } = child.props;
+          const active = activeFilters[accessor];
+          const { filters: data } = options.find(option => option.section === accessor) || {};
+
+          if (!data) {
+            return <div>Filter with section: {accessor} not found.</div>;
+          }
+
+          return React.cloneElement(child, {
+            active,
+            data,
+            isOpen: this.state.openFilter === accessor && !this.state.forceCollapse,
+            onHeaderClick: () => {
```

Comment:
`onHeaderClick` and `onChange` are going to generate a new function on every render. This is bad for performance.. https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-no-bind.md

Maybe try to find a way to refactor this?

Reply:
Out of ideas for the moment